src/meta/src/hummock/compaction/picker/ttl_reclaim_compaction_picker.rs

```diff
@@ -48,30 +115,84 @@ impl TtlReclaimCompactionPicker {
         let mut select_input_ssts = vec![];
         let level_handler = &level_handlers[reclaimed_level.level_idx as usize];
 
-        if state.last_select_index >= reclaimed_level.table_infos.len() {
-            state.last_select_index = 0;
+        if reclaimed_level.table_infos.is_empty() {
```

Comment:
ditto: it is better to avoid duplicated codes with space reclaim picker since the differences are only in `self.filter(...)`